packages/fluentui/react-builder/src/components/NarrationComputer.tsx

```diff
@@ -0,0 +1,268 @@
+/*
+ TODO:
+* Add the missing and not so obvious attributes (e.g. "aria-haspopup" or "aria-expanded") for the already defined roles (e.g. "menuitem" or "checkbox") according to the specification.
+* With JAWS, in the case of the element with the "listbox" role, differentiate between having and not having the aria-multiselectable="true" attribute. If this attribute is present, then aria-selected="false" on the child elements with the role "option" behave differently than if it is not present. Specifically, if aria-multiselectable="true" is present, the aria-selected="false" causes the narration of "not selected", but if present, having the aria-selected attribute makes no difference to the narration. 
+ * Should we also consider the "disabled" state?
+ */
+import SRNC from './SRNC-Definitions';
+import './SRNC-Rules-Win_JAWS';
+
+export default class NarrationComputer {
+  private computedParts: Record<string, string> = {
+    value: '',
+    name: '',
+    description: '',
+    type: '',
+    state: '',
+    position: '',
+    usage: '',
+  };
+
+  // Computes and returns the complete screen reader narration for the given element and platform.
+  async computeNarration(element: Element, platform: string): Promise<string> {
+    let definitionName = this.getDefinitionName(element, platform, 'stateRules');
+
+    // Retrieve the computed accessible node
+    const node = await (window as any).getComputedAccessibleNode(element);
+
+    // Compute and store all the narration parts
+    this.computeUsage(definitionName, element, platform);
+    this.computeDescription(definitionName, element, platform);
+    this.computeNameAndTitle(node, element);
+    this.computeValue(node);
+    this.computePosition(definitionName);
+    this.computeTypeAndState(node, definitionName, element, platform);
+
+    definitionName = this.getDefinitionName(element, platform, 'readingOrder');
+    const computedNarration = this.composeNarrationFromParts(definitionName, platform);
+    return computedNarration;
+  } // End computeNarration
+
+  // Returns the definition name based on the given element, platform and definition type.
+  getDefinitionName(element: Element, platform: string, definitionType: string): string {
+    // Determine the definitions source by the definition type
+    let definitions;
+    if (definitionType === 'readingOrder') {
+      // Begin if 1
+      definitions = SRNC.readingOrder[platform];
+    } else if (definitionType === 'stateRules') {
+      // else if 1
+      definitions = SRNC.stateRules[platform];
+    } // End if 1
```

Comment:
```suggestion
    const definitionTypes = {
        readingOrder: SRNC.readingOrder[platform],
        stateRules: SRNC.stateRules[platform]
    }
    let definitions = definitionTypes[definitionType];
```

Not sure how big `definitionTypes` can grow so I guess avoiding the if/else would be good thing here.

Reply:
I don't expect it to grow but I can apply this suggestion for the simplicity sake.

Reply:
So the compiler complains:

```
Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{ readingOrder: any; stateRules: any; }'.
  No index signature with a parameter of type 'string' was found on type '{ readingOrder: any; stateRules: any; }'.  TS7053
```
 
I had to change it to:

```
const definitionTypes: Record<string, any> = {
```

Reply:
I am good with it :)